docs/source/deep-learning/pytorch/guide/index.rst

```diff
@@ -0,0 +1,298 @@
+practices that we recommend you to follow:
+
+- Log your model and training parameters via :py:func:`mlflow.log_params()` at the beginning of training
+  loop, such as learning rate, batch size, etc. :py:func:`mlflow.log_params()` is the
+  batched logging version of :py:func:`mlflow.log_param()`, which is more efficient than the latter.
+- Log your model architecture at the begnning of training via :py:func:`mlflow.log_artifact()`. You can use
+  ``torchinfo`` package to get the model summary.
+- Log training and validation metrics via :py:func:`mlflow.log_metric()` inside your training loop, such as
+  loss and accuracy for classification tasks. If you have multiple metrics per logging step, you can
+  use :py:func:`mlflow.log_metrics()` to log them together.
+- Log your trained/finetuned model to MLflow via `mlflow.pytorch.log_model()` at the end of your training.
+- [Optional] You can also log your model checkpoints to MLflow via :py:func:`mlflow.log_artifact()` during
+  training if you wish to keep middle training status.
+
+The following is an end-to-end example of how to log your PyTorch experiments to MLflow:
+
+.. code-block:: python
+
+    import mlflow
+    import torch
+
+    from torch import nn
+    from torch.utils.data import DataLoader
+    from torchinfo import summary
+    from torchmetrics import Accuracy
+    from torchvision import datasets
+    from torchvision.transforms import ToTensor
+
+    # Download training data from open datasets.
+    training_data = datasets.FashionMNIST(
+        root="data",
+        train=True,
+        download=True,
+        transform=ToTensor(),
+    )
+
+    # Create data loaders.
+    train_dataloader = DataLoader(training_data, batch_size=64)
+
+    # Get cpu or gpu for training.
+    device = "cuda" if torch.cuda.is_available() else "cpu"
+
+
+    # Define the model.
+    class NeuralNetwork(nn.Module):
+        def __init__(self):
+            super().__init__()
+            self.flatten = nn.Flatten()
+            self.linear_relu_stack = nn.Sequential(
+                nn.Linear(28 * 28, 512),
+                nn.ReLU(),
+                nn.Linear(512, 512),
+                nn.ReLU(),
+                nn.Linear(512, 10),
+            )
+
+        def forward(self, x):
+            x = self.flatten(x)
+            logits = self.linear_relu_stack(x)
+            return logits
+
+
+    def train(dataloader, model, loss_fn, metrics_fn, optimizer):
+        model.train()
+        for batch, (X, y) in enumerate(dataloader):
+            X, y = X.to(device), y.to(device)
+
+            pred = model(X)
+            loss = loss_fn(pred, y)
+            accuracy = metrics_fn(pred, y)
+
+            # Backpropagation.
+            loss.backward()
+            optimizer.step()
+            optimizer.zero_grad()
+
+            if batch % 100 == 0:
+                loss, current = loss.item(), batch
+                mlflow.log_metric("loss", f"{loss:3f}", step=(batch // 100))
```

Comment:
Can we pass a string into log_metric? The documentation says the value should be a float: https://mlflow.org/docs/latest/python_api/mlflow.html#mlflow.log_metric

Reply:
yea we can. I don't think we should say it is a float, basically it can be everything, will update the docstring accordingly.